src/main/java/com/redislabs/redistimeseries/Range.java

```diff
@@ -24,4 +29,84 @@ public String getKey() {
   public Value[] getValues() {
     return values;
   }
+
+  protected static Value[] parseRange(List<Object> range) {
+    Value[] values = new Value[range.size()];
+
+    for(int i=0; i<values.length ; ++i) {
+      @SuppressWarnings("unchecked") List<Object> tuple = (List<Object>)range.get(i);
+      values[i] = Value.parseValue(tuple);
+    }
+    return values;
+  }
+
+  protected static Range[] parseRanges(List<?> result) {
+    Range[] ranges = new Range[result.size()];
+    for(int j=0; j<ranges.length; ++j) {
+      List<?> series = (List<?>)result.get(j);
+      String resKey = SafeEncoder.encode((byte[])series.get(0));
+      List<?> resLabels = (List<?>)series.get(1);
+      Map<String, String> rangeLabels = new HashMap<>();
+      for (Object resLabel : resLabels) {
+        List<byte[]> label = (List<byte[]>) resLabel;
+        rangeLabels.put(SafeEncoder.encode(label.get(0)), SafeEncoder.encode(label.get(1)));
+      }
+      Value[] values = parseRange((List<Object>) series.get(2));
+      ranges[j] = new Range(resKey, rangeLabels, values);
+    }
+    return ranges;
+  }
+
+  protected static Range[] parseMget(List<?> result) {
+    Range[] ranges = new Range[result.size()];
+    for(int j=0; j<ranges.length; ++j) {
+      List<?> series = (List<?>)result.get(j);
+      String resKey = SafeEncoder.encode((byte[])series.get(0));
+      List<?> resLabels = (List<?>)series.get(1);
+      Map<String, String> rangeLabels = new HashMap<>();
```

Comment:
Map<String, String> rangeLabels = new HashMap<>(resLabels.size());

Reply:
If we are already doing some code refactor perhaps you should make a function of 

```java
Map<String, String> rangeLabels = new HashMap<>(resLabels.size());
for (Object resLabel : resLabels) {
        List<byte[]> label = (List<byte[]>) resLabel;
        rangeLabels.put(SafeEncoder.encode(label.get(0)), SafeEncoder.encode(label.get(1)));
}
```

Reply:
agree. doing so